src/Features/Core/Portable/IncrementalCaches/SymbolTreeInfoIncrementalAnalyzerProvider.cs

```diff
@@ -270,7 +296,7 @@ private Task UpdateReferencesAync(Project project, CancellationToken cancellatio
                     // We still want to cache that result so that don't try to continuously produce
                     // this info over and over again.
                     metadataInfo = new MetadataInfo(info, metadataInfo.ReferencingProjects ?? new HashSet<ProjectId>());
-                    _metadataPathToInfo.AddOrUpdate(key, metadataInfo, (_1, _2) => metadataInfo);
+                    _metadataPathToInfo[key] = metadataInfo;
```

Comment:
well that's much nicer.  possibly worth auditing the codebase to see where i would  have made this same mistake elsewhere.

Reply:
Opened https://github.com/dotnet/roslyn/issues/33179 to track.